src/backend/cuda/nvrtc/kernel_map.hpp

```diff
@@ -0,0 +1,55 @@
+/*******************************************************
+ * Copyright (c) 2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <nvrtc_kernel_headers/convolve1.hpp>
+#include <nvrtc_kernel_headers/convolve2.hpp>
+#include <nvrtc_kernel_headers/convolve3.hpp>
+#include <nvrtc_kernel_headers/convolve_separable.hpp>
+#include <nvrtc_kernel_headers/scan_dim.hpp>
+#include <nvrtc_kernel_headers/scan_first.hpp>
+#include <nvrtc_kernel_headers/scan_dim_by_key.hpp>
+#include <nvrtc_kernel_headers/scan_first_by_key.hpp>
+#include <nvrtc_kernel_headers/transpose.hpp>
+#include <nvrtc_kernel_headers/where.hpp>
+
+#include <map>
+#include <string>
+
+namespace cuda {
+
+/**
+    \brief A map of CUDA kernel source codes
+ */
+static const std::map<std::string, std::string> KernelMap = {
+    { "transpose", std::string(transpose_cuh, transpose_cuh_len) },
```

Comment:
Thinking about this again. Maybe we should add the headers in the same source file. Like this:

```
KernelTemplate transposeTemplate = MakeKernel(transpose_cuh, transpose_cuh_len);
...
void transpose(...) {
    auto transpose = transposeTemplate(
                            {TemplateArgs(), ...},
                            {Defines(), ....}
                        );
}
```